modules/table/src/integrationTest/java/org/apache/ignite/distributed/ItInternalTableScanTest.java

```diff
@@ -264,14 +264,14 @@ public void testMultipleRowScan() throws Exception {
      */
     @Test()
     public void testInvalidRequestedAmountScan() throws Exception {
-        AtomicBoolean cursorClosed = new AtomicBoolean(false);
+        CyclicBarrier cursorClosedBarrier = new CyclicBarrier(2);
```

Comment:
I think Atomic is enough here.
Because when an exception is already having, the invocation is guaranteed happened.

Reply:
I see these are different waiting: one of the waiting of internal cursor will be closed, another awaiting about the notification process completed.

Reply:
Was discussed explicitly.